src/librustc_mir/transform/const_prop.rs

```diff
@@ -890,6 +906,12 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
                     );
                     Self::remove_const(&mut self.ecx, place.local);
                 }
+            } else {
```

Comment:
This arm didn't exist before, it's the `else` for the `layout_of` check. If we can't compute the layout, erase anything that's in that local, even if I don't think that it can actually happen that there's something in the local.